java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_1/Number1.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    public void setNumberOneDouble(double numberOneDouble) {
+        this.numberOneDouble = numberOneDouble;
+    }
+
+    public void setNumberTwoDouble(double numberTwoDouble) {
+        this.numberTwoDouble = numberTwoDouble;
+    }
+
+    public void addition() {
+        if (!isInteger) {
+            calculationForDouble = numberOneDouble + numberTwoDouble;
+            System.out.println("Result:" + " " + calculationForDouble);
+        }
+        else {
+            calculationForInteger = numberOne + numberTwo;
+            System.out.println("Result:" + " " + calculationForInteger);
+        }
+    }
+
+    public void subtraction() {
+        if (!isInteger) {
+            calculationForDouble = numberOneDouble - numberTwoDouble;
+            System.out.println("Result:" + " " + calculationForDouble);
+        }
+        else {
+            calculationForInteger = numberOne - numberTwo;
+            System.out.println("Result:" + " " + calculationForInteger);
+        }
+    }
+
+    public void multiplication() {
+        if (!isInteger) {
+            calculationForDouble = numberOneDouble * numberTwoDouble;
+            System.out.println("Result:" + " " + calculationForDouble);
+        }
+        else {
+            calculationForInteger = numberOne * numberTwo;
+            System.out.println("Result:" + " " + calculationForInteger);
+        }
+    }
+
+    public void division() {
+        if (!isInteger) {
+            if (numberOneDouble == 0.0) {
+                System.out.println("Zero is not dividable\n");
+                number1.calculations();
+            } else {
+                try {
+                    calculationForDouble = numberOneDouble / numberTwoDouble;
+                    System.out.println("Result:" + " " + calculationForDouble);
+                } catch (ArithmeticException error) {
+                    System.out.println("Zero is not dividable\n");
+                    number1.calculations();
+                }
+            }
+        } else {
+            if (numberOne == 0) {
+                System.out.println("Zero is not dividable\n");
+                number1.calculations();
+            } else {
+                try {
+                    calculationForInteger = numberOne / numberTwo;
+                    System.out.println("Result:" + " " + calculationForInteger);
+                } catch (ArithmeticException error) {
+                    System.out.println("Zero is not dividable\n");
+                    number1.calculations();
+                }
+            }
+        }
+    }
+}
+
+class RangeOfNumber {
+
+    Calculator calculator = new Calculator();
+
+    boolean outOfRange;
+
+    public void rangeOfNumberInt(int number1, int number2) {
+        int resultOfAddition = number1 + number2;
+        int resultOfMultiplication = number1 * number2;
+
+        if (resultOfAddition == Integer.MAX_VALUE || resultOfMultiplication == Integer.MAX_VALUE) {
```

Comment:
1. What if the result is bigger than Integer.MAX_VALUE? In other words, it actually wraparounds.
2. Why don't we check subtraction? It can overflow as well.